### PR TITLE
Us67 sign up

### PR DESCRIPTION
**Pull request by vcedgar**

### CODE REVIEW

**Review by majluni — COMMENTED** [review]
Can we print some type of message as to why the login or create do not work? Like, if i try to create an empty username and password then print what the requirements for them are

File: package.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
   "dependencies": {
     "electron-compile": "^6.4.4",
     "electron-squirrel-startup": "^1.0.0",
+    "mkdirp": "^0.5.1",
```
Comment (zlmonroe) [inline package.json:52]: Is this dependency actually needed?

File: package-lock.json [hunk 2]
```diff
@@ -5,9 +5,9 @@
   "requires": true,
```
Comment (zlmonroe) [inline package-lock.json:5]: There is a whole lot of changes to the package lock here..... are these needed for everyone?

File: spec/UserDbSpec.js [hunk 3]
```diff
@@ -2,20 +2,47 @@
 /* global it */
```
Comment (zlmonroe) [inline spec/UserDbSpec.js:2]: This doesn't pass the linter. I am betting other things fail too (I won't bother leaving a bunch of comments, just one as a note)

File: spec/UserDbSpec.js [hunk 4]
```diff
@@ -2,20 +2,47 @@
...
+        expect(result3).toBe(false);
+    });
 
     /* A specification */
-    it("shall be able to create a default user with role: student", async function() {
-        const userDb = UserDB;
-        userDb.addUser("test", "test");
-        const foundUser = await userDb.userExists("test", null);
-        expect(foundUser).toBe(true)
+    it("shall be able to add a user with a new username and fail at adding a user with an existing username", async function() {
```
Comment (zlmonroe) [inline spec/UserDbSpec.js]: You shouldn't have "and"s in specs

**Review by zlmonroe — CHANGES_REQUESTED** [review]
Everything seems to run fine. Mostly just fix the lint style and separate the spec

File: spec/UserDbSpec.js (hunk 4 above)
Comment (vcedgar) [reply spec/UserDbSpec.js]: This means I will have to copy and paste nearly the exact same test because they are related :/

Comment (zlmonroe) [reply spec/UserDbSpec.js]: Just changed the spec name then. Something like "shall add new users only when their username does not already exist"

Comment (vcedgar) [reply spec/UserDbSpec.js]: Done

File: spec/UserDbSpec.js (hunk 3 above)
Comment (vcedgar) [reply spec/UserDbSpec.js:2]: Done!

File: package-lock.json (hunk 2 above)
Comment (vcedgar) [reply package-lock.json:5]: Added package lock to ignore

File: package.json (hunk 1 above)
Comment (vcedgar) [reply package.json:52]: Needed it for smth :/ Added this to git ignore

**Review by zlmonroe — APPROVED** [review]

**Pull request merged**